internal/app/secretless/handlers/mysql/authentication_handshake.go

```diff
@@ -0,0 +1,380 @@
+		 | HandshakeResponse             |                                     |
+		 |------------------------------>|                                     |
+		 |                               | -------------------------------\    |
+		 |                               |-| If client requested SSL, add |    |
+		 |                               | | it to HandshakeResponse      |    |
+		 |                               | |------------------------------|    |
+		 |                               | HandshakeResponse                   |
+		 |                               |------------------------------------>|
+		 |                               |                                     |
+		 |                               |                            OkPacket |
+		 |                               |<------------------------------------|
+		 |                               |                                     |
+		 |                      OkPacket |                                     |
+		 |<------------------------------|                                     |
+		 |                               |                                     |
+
+Note: The above diagram was created using https://textart.io/sequence and the
+following source:
+
+	object Client Secretless Backend
+	Backend->Secretless: InitialHandshakePacket
+	Secretless->Client: InitialHandshakePacket
+	Client->Secretless: HandshakeResponse
+	note right of Secretless: If client requested SSL, add\n it to HandshakeResponse
+	Secretless->Backend: HandshakeResponse
+	Backend->Secretless: OkPacket
+	Secretless->Client: OkPacket
+
+*/
+type AuthenticationHandshake struct {
+	connectionDetails *ConnectionDetails
+	clientConn        *Connection
+	backendConn       *Connection
+
+	sslMode *ssl.DbSSLMode
+
+	//TODO: after kumbi's work these 2 should be combined
+	rawServerHandshake      Packet
+	serverHandshake         *protocol.HandshakeV10
+	clientHandshakeResponse *protocol.HandshakeResponse41
+
+	err               error
+}
+
+// NewAuthenticationHandshake creates a new AuthenticationHandshake command object,
+// intended to be Run().
+//
+func NewAuthenticationHandshake(
+	clientConn *Connection,
+	backendConn *Connection,
+	connDetails *ConnectionDetails,
+) AuthenticationHandshake {
+	return AuthenticationHandshake{
+		connectionDetails: connDetails,
+		clientConn:        clientConn,
+		backendConn:       backendConn,
+	}
+}
+
+// Run executes all the logic needed to complete authentication between a
+// MySQL server and client.  When it completes successfully,
+// AuthenticatedBackendConn will return the raw, authenticated network conn.
+//
+func (h *AuthenticationHandshake) Run() error {
+	h.readServerHandshake()
+	h.writeHandshakeToClient()
+	h.validateServerSSL()
+	h.readClientHandshakeResponse()
+	h.overrideClientCapabilities()
+	h.injectCredentials()
+	h.handleClientSSLRequest()
+	h.writeClientHandshakeResponseToBackend()
+
+	// TODO: Kumbi: what does this mean? AuthSwitchRequest, AuthSwitchResponse
```

Comment:
is this getting left in still? looking at the code without any context, I'm not sure why this is here

Reply:
removed